### ISSUE TITLE
Standalone works but plugin always timeouts when attempting to load a windows dll.

### ISSUE THREAD

**Issue by JohnnyKeys1** [falkTX/Carla#1265]
The title describes this problem: Carla standalone functions perfectly, but the plugin does not. Some research reveals that when this has happened before it has been because of 'bridge version not matching carla version' however I cannot find a solution.
This occurs for all dlls, including free, common vst's such as GSnap.
The timeout is always instant - as if it not even 'trying'.
Thankyou for this wonderful program I am so happy to trash MS OS.

System:
Linux Mint 20 with low latency kernel and everything else working as intended.

**Comment by falkTX** [comment]
This usually happens when there is indeed a version mismatch.
I added a way for the bridges to dynamically change their window title names, which maybe unintentionally broke something.
In any case, 2.2 will be out this weekend, so just update everything after you see the announcement, and the versions will be in sync again.

**Comment by JohnnyKeys1** [comment]
Thats great to hear!
Thankyou you are a rockstar of linux audio :+1:

**Comment by JohnnyKeys1** [comment]
so uhhhh, did 2.2 come out? I ask because its still not working...

**Comment by JohnnyKeys1** [comment]
bump

**Comment by JohnnyKeys1** [comment]
Hi,
please forgive my persistence! 
Today, the app was updated to 2.2 like you said it would be, but unfortunately the same issue exists, the bridge immediately times out.

**Comment by JohnnyKeys1** [comment]
Everyday at breakfast I check this forum... hoping for some kind of answer...

**Comment by 1b22b** [comment]
I think I may be having the same or a similar problem. I have a Ryzen 5 processor and plenty of RAM, so I'm not sure why it says it's timing out loading a dll file.

When I try to open any Windows VST in either the stand-alone or VST interface in Ardour, it says "Timeout while waiting for a response from plugin-bridge (or the plugin crashed on initialization?)"

Things had been working fine until about a month ago and now I can't even load some projects because it won't load Carla and my Windows VSTs.

**Comment by JohnnyKeys1** [comment]
I hear you brother, its not just you. The bastion of linux audio freedom will prevail because of our forum posts.

PS Breakfast this morning was a hearty weetbix whilst checking this post for an update :D

**Comment by JohnnyKeys1** [comment]
Here's me posting here again to just say I'm so excited at the anticipated possibility of being able to open the projects I was able to open before Carla was broken 2 'updates' ago.

**Comment by falkTX** [comment]
well, lets get some info on this...

1. did you change the carla wine settings so that "wine" executable no longer works?
2. are you carla to run the plugin through carla-single? (run it without arguments to see examples of its usage)
3. are wine programs able to access Z: drive as the linux root?

**Comment by 1b22b** [comment]
1. I didn't touch any settings on anything. I had enough trouble installing the software (I am not a Linux or programming expert) to know better than to go around tinkering with things. All I do is download the updates for Carla whenever Ubuntu tells me that some are available, and that alone was enough to break it.
2. Not sure what carla-single is, but if you mean the stand-alone program outside of my DAW (which is Ardour), then that doesn't work either. It gives me the same message about timing out after about 1 second of trying to load.
3. I wouldn't even know how to check that.

Really there are fewer and fewer Windows-only VSTs that I like, and if I have to I can just boot a virtual machine and do some minor editing in a Windows 7 VM before moving the files back into Ardour and using only Linux VSTs. I know that Microsoft and its money are reaching deeper and deeper into the Linux community anyway so maybe I should just stop all updates while my computer is still functional.

**Comment by falkTX** [comment]
2. carla-single is something you run on a terminal/console, assuming you are using JACK.

3. if you just run "wine explorer" you can open the wine file explorer, from there you can see if Z: is accessible.

Nothing personal against you, but if you prefer to keep software static and not update, then dont report issues as well.
Whatever broke was not intentional, and likely not even within Carla.

**Comment by 1b22b** [comment]
Yes, Z is accessible from there, though I noticed something interesting in the process. When I first tried to run "wine explorer," I got this:

"wine: '/home/brian/.wine' is a 64-bit installation, it cannot be used with a 32-bit wineserver."

So I followed some instructions online and deleted my .wine folder, so it had to reconfigure itself and then it loaded.

I tried running Carla again by itself and in Ardour after doing that with wine, but same result: an error message saying "Timeout while waiting for a response from plugin-bridge (or the plugin crashed on initialization?)." It does this for any and all Windows .vst files I try to load.


When I run "carla-single" in terminal I get this:

carla-single
usage: /usr/bin/carla-single [arch (optional)] [format] [filename/uri] [label (optional)] [uniqueId (optional)]

Possible archs:
  - native (default)
  - linux32
  - linux64
  - win32
  - win64

Possible formats:
  - internal
  - ladspa
  - dssi
  - lv2
  - vst|vst2
  - vst3
  - sf2
  - sfz

Examples:
$ /usr/bin/carla-single internal midisplit
$ /usr/bin/carla-single dssi /usr/lib/dssi/whysynth.so
$ /usr/bin/carla-single lv2 http://calf.sourceforge.net/plugins/Compressor
$ /usr/bin/carla-single native vst /usr/lib/vst/TAL-NoiseMaker.so
$ /usr/bin/carla-single win32 vst "~/.wine/drive_c/Program Files (x86)/VstPlugins/Kontakt 5.dll"
$ CARLA_BRIDGE_PLUGIN_BINARY_TYPE=win64 /usr/bin/carla-single vst "~/.wine/drive_c/Program Files/Common Files/VST2/Sytrus VSTi.dll"

Format can be skipped for internal and lv2 plugins (auto-detected), like this:
$ /usr/bin/carla-single midipattern
$ /usr/bin/carla-single http://synthv1.sourceforge.net/lv2


I haven't been keeping software static on Ubuntu so far, though I was on Linux Mint after they switched from Wine to that non-working "Play on Linux." Also I didn't report this issue though I am happy to help move it along if possible as I'm sure we're not unique.
